libnestutil/regula_falsi.h

```diff
@@ -0,0 +1,112 @@
+/*
+ *  regula_falsi.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef REGULA_FALSI_H
+#define REGULA_FALSI_H
+
+
+namespace nest
+{
+
+/**
+ * Localize threshold crossing by using Illinois algorithm of regula falsi method.
+ *
+ * See https://en.wikipedia.org/wiki/Regula_falsi#The_Illinois_algorithm for details on the algorithm.
+ *
+ * @param   Node   model object that call function
+ * @param   function threshold_distance function
+ * @param   double length of interval since previous event
+ * @returns time from previous event to threshold crossing
+ */
+template < typename CN >
+double
+regula_falsi( CN* node, double ( Node::*threshold_distance )( double ) const, const double dt )
```

Comment:
@stinebuu Nice, and apologies for having complicated things more than necessary, as I had forgotten about the details of pointer to member function (too much Python programming lately ...). In contrast to Python, C++ does not have bound member functions, so we need to point the node anyways, and then we could just pass the node:

```c++
template < typename CN >
double 
regula_falsi( const CN& node, double dt )
...
```